mountpoint-s3/src/fs.rs

```diff
@@ -804,6 +804,7 @@ impl From<InodeError> for i32 {
             InodeError::CannotRemoveRemoteDirectory(_) => libc::EPERM,
             InodeError::DirectoryNotEmpty(_) => libc::ENOTEMPTY,
             InodeError::UnlinkNotPermittedWhileWriting(_) => libc::EPERM,
+            InodeError::CorruptedMetadata(_, _) => libc::EPERM,
```

Comment:
Not sure if there is a more fitting error.

Reply:
I think maybe `EIO` is slightly more appropriate, but it's pretty niche either way.